x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -101,10 +104,37 @@ const AssistantComponent: React.FC<Props> = ({
 
   // Welcome conversation is a special 'setup' case when no connector exists, mostly extracted to `ConnectorSetup` component,
   // but currently a bit of state is littered throughout the assistant component. TODO: clean up/isolate this state
-  const welcomeConversation = useMemo(
-    () => conversations[selectedConversationId] ?? BASE_CONVERSATIONS[WELCOME_CONVERSATION_TITLE],
-    [conversations, selectedConversationId]
-  );
+  const welcomeConversation = useMemo(() => {
+    const conversation =
+      conversations[selectedConversationId] ?? BASE_CONVERSATIONS[WELCOME_CONVERSATION_TITLE];
+    const doesConversationHaveMessages = conversation.messages.length > 0;
+    if (!isAssistantEnabled) {
+      if (
+        !doesConversationHaveMessages ||
+        conversation.messages[conversation.messages.length - 1].content !==
+          enterpriseMessaging[0].content
+      ) {
+        return {
+          ...conversation,
+          messages: [...conversation.messages, ...enterpriseMessaging],
```

Comment:
An an unlicensed state, this is where the enterprise messaging is appended to the end of the conversation, because the last message does not include that content

Reply:
nit: only messages is varying in that `if..else` checks, so object can be moved out of it
```
{
          ...conversation,
```